packages/azure-kusto-ingest/src/abstractKustoClient.ts

```diff
@@ -45,4 +49,41 @@ export abstract class AbstractKustoClient {
             throw new Error("Client is closed");
         }
     }
+
+    getIngestionEndpoint(clusterUrl?: string): string | undefined {
+        if (!clusterUrl || clusterUrl.includes(INGEST_PREFIX) || this.isReservedHostname(clusterUrl)) {
+            return clusterUrl;
+        }
+        return clusterUrl.replace(PROTOCOL_SUFFIX, PROTOCOL_SUFFIX + INGEST_PREFIX);
+    }
+
+    getQueryEndpoint(clusterUrl?: string): string | undefined {
+        if (clusterUrl && clusterUrl.includes(INGEST_PREFIX)) {
+            return clusterUrl.replace(INGEST_PREFIX, "");
+        }
+        return clusterUrl;
+    }
+
+    isReservedHostname(clusterUrl: string): boolean {
+        const parsedUrl = new URL(clusterUrl);
+        const authority = parsedUrl.hostname;
+        if (!authority) {
+            return true;
+        }
+        let is_ip;
```

Comment:
Obviously I don't like this.

You could have used the Address4.IsValid static methods for this.

But since we only need to check /if/ it's an ip, we can just use this package - https://www.npmjs.com/package/is-ip, which has a very simple api.

Reply:
Address4.IsValid is deprected, and is-ip doesnt support browser. ended up using this module instead

Reply:
It does

Reply:
oops. thought you meant a different one